### ISSUE TITLE
User specified replication factor is ignored

### ISSUE THREAD

**Issue by splunkbot9000** [splunk/splunk-ansible#248]
I have my replication factor set to 2 and search factor to 2 however the fact is being set to 1 for both at runtime. I'm trying this on a 2 node indexer cluster and I have a branch (derived from the 7.3.1 tag) which totally removes [the play which sets this to 1 if there are less than 3 indexers](https://github.com/splunk/splunk-ansible/blob/fdb5708be330e0cae34b955aec7a0a57bb2d3c92/roles/splunk_cluster_master/tasks/initialize_cluster_master.yml#L11). The output from $ python /opt/ansible/inventory/environ.py shows the correct values of 2.

ansible verbose output
>2019-08-28 19:48:30,919 p=1481 u=root |  Wednesday 28 August 2019  19:48:30 -0500 (0:00:00.074)       0:00:12.404 ******
>2019-08-28 19:48:30,953 p=1481 u=root |  TASK [splunk_cluster_master : Get default replication factor] ******************
>2019-08-28 19:48:30,953 p=1481 u=root |  task path: /opt/ansible/roles/splunk_cluster_master/tasks/initialize_cluster_master.yml:6
>2019-08-28 19:48:30,954 p=1481 u=root |  ok: [localhost] => {
>    "ansible_facts": {
>        "idxc_replication_factor": "1",
>        "idxc_search_factor": "1"
>    },
>    "changed": false
>}
.....
>2019-08-28 19:48:32,340 p=1481 u=root |  TASK [splunk_cluster_master : Set the current node as a Splunk indexer cluster master] ***
>2019-08-28 19:48:32,340 p=1481 u=root |  task path: /opt/ansible/roles/splunk_cluster_master/tasks/initialize_cluster_master.yml:31
>2019-08-28 19:48:32,347 p=1481 u=root |  changed: [localhost] => {
>    "attempts": 1,
>    "changed": true,
>    "cmd": [
>        "/opt/splunk/bin/splunk",
>        "edit",
>        "cluster-config",
>        "-mode",
>        "master",
>        "-replication_factor",
>        "1",
>        "-search_factor",
>        "1",
>        "-secret",
>        "<REDACTED>",
>        "-cluster_label",
>        "<REDACTED>",
>        "-auth",
>        "admin:<REDACTED>"
>    ],
>    "delta": "0:00:00.642018",
>    "end": "2019-08-28 19:48:32.287718",
>    "invocation": {
>        "module_args": {
>            "_raw_params": "/opt/splunk/bin/splunk edit cluster-config -mode master -replication_factor 1 -search_factor 1 -secret '<REDACTED>' -cluster_label '<REDACTED>' -auth 'admin:<REDACTED>'",
>            "_uses_shell": false,
>            "argv": null,
>            "chdir": null,
>            "creates": null,
>            "executable": null,
>            "removes": null,
>            "stdin": null,
>            "stdin_add_newline": true,
>            "strip_empty_ends": true,
>            "warn": true
>        }
>    },
>    "rc": 0,
>    "start": "2019-08-28 19:48:31.645700"
>}

idxc section in /tmp/defaults/default.yml
```
  idxc:
    enable: true
     label: REDACTED
     replication_factor: 2
     replication_port: 9887
     search_factor: 2
     secret: REDACTED
```

Resulting clustering stanza in server.conf
```
[clustering]
cluster_label = REDACTED
mode = master
pass4SymmKey = REDACTED
replication_factor = 1
search_factor = 1
```

Any ideas to enforce the values I've set in my default.yml are greatly appreciated!

**Comment by schose** [comment]
could you verify the whitespaces? for me it's working..

```
      idxc:
        enable: true
        replication_port: 9887
        search_factor: 1
        replication_factor: 2
        label: idxc_label
```

**Comment by splunkbot9000** [comment]
> could you verify the whitespaces? for me it's working..
> 
> ```
>       idxc:
>         enable: true
>         replication_port: 9887
>         search_factor: 1
>         replication_factor: 2
>         label: idxc_label
> ```

I have two whitespaces before idxc: and two more for the idxc parameters

```
---
ansible_post_tasks: null
ansible_pre_tasks: null
hide_password: false
retry_num: 50
shc_bootstrap_delay: 30
splunk_home_ownership_enforcement: true

config:
  baked: default.yml
..

splunk:
..
  idxc:
    enable: true
    label: REDACTED
    replication_factor: 2
    replication_port: 9887
    search_factor: 2
    secret: REDACTED
```

**Comment by nwang92** [comment]
Sorry for the lack of response - regarding the "Lower indexer search/replication factor" task, is this something you'd like to see removed? I'm not entirely sure of the history of this task, but maybe it can instead be something akin to `min(3, num_indexer_hosts)`?

**Comment by nwang92** [comment]
Closing this issue since I made a PR to address the case of <=2 indexers/search heads when clustering each tier. This will be available in the next release of Splunk (8.0.2)
